controllers/machinedeployment_sync_test.go

```diff
@@ -397,3 +398,109 @@ func TestScaleMachineSet(t *testing.T) {
 		})
 	}
 }
+
+func newDeploymentStatus(replicas, updatedReplicas, availableReplicas int32) clusterv1.MachineDeploymentStatus {
+	return clusterv1.MachineDeploymentStatus{
+		Replicas:          replicas,
+		UpdatedReplicas:   updatedReplicas,
+		AvailableReplicas: availableReplicas,
+	}
+}
+
+func currentDeployment(pds *int32, replicas, statusReplicas, updatedReplicas, availableReplicas int32, conditions clusterv1.Conditions) *clusterv1.MachineDeployment {
+	d := &clusterv1.MachineDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "progress-test",
+		},
+		Spec: clusterv1.MachineDeploymentSpec{
+			ProgressDeadlineSeconds: pds,
+			Replicas:                &replicas,
+			Strategy: &clusterv1.MachineDeploymentStrategy{
+				Type: clusterv1.RollingUpdateMachineDeploymentStrategyType,
+				RollingUpdate: &clusterv1.MachineRollingUpdateDeployment{
+					MaxUnavailable: intOrStrPtr(0),
+					MaxSurge:       intOrStrPtr(1),
+					DeletePolicy:   pointer.StringPtr("Oldest"),
+				},
+			},
+		},
+		Status: newDeploymentStatus(statusReplicas, updatedReplicas, availableReplicas),
+	}
+	d.Status.Conditions = conditions
+	return d
+}
+
+// helper to create MS with given availableReplicas.
+func newMSWithAvailable(name string, specReplicas, statusReplicas, availableReplicas int32) *clusterv1.MachineSet {
+	return &clusterv1.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              name,
+			CreationTimestamp: metav1.Time{},
+			Namespace:         metav1.NamespaceDefault,
+		},
+		Spec: clusterv1.MachineSetSpec{
+			Replicas: pointer.Int32Ptr(specReplicas),
+		},
+		Status: clusterv1.MachineSetStatus{
+			Selector:          "",
+			AvailableReplicas: availableReplicas,
+			Replicas:          statusReplicas,
+		},
+	}
+}
+
+func TestMachineDeploymentSyncDeploymentStatus(t *testing.T) {
```

Comment:
This should be named `TestSyncDeploymentStatus`

Reply:
Done.